src/data_access/MemoryUserDataAccessObject.java

```diff
@@ -0,0 +1,57 @@
+package data_access;
+
+import entity.NutrientRange;
+import entity.User;
+import entity.UserPreference;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class MemoryUserDataAccessObject implements LoginUserDataAccessInterface,
+                                                   SignupDataAccessInterface,
+                                                   SavePreferencesDataAccessInterface {
+    private final Map<String, User> accounts = new HashMap<>();
+
+    /**
+     * Saves user to database
+     * @param user represents the user to save
+     */
+    @Override
+    public void saveUser(User user) {
+        accounts.put(user.getUsername(), user);
+    }
+
+    /**
+     * Saves user preferences to database
+     * @param username represents the username of the user to save to
+     * @param preference represents the new preference
+     */
+    @Override
+    public void saveUserPreference(String username, UserPreference preference) {
+        User user = accounts.get(username);
+        user.setPreferences(preference);
```

Comment:
Might not be necessary for our MVP, but might need to add either some kind of pre-condition stating that username must be the username of a User already saved to accounts, or an if statement that throws an error / exception if username isn't the username of a User in accounts.

Reply:
Yes I agree. I think if we have time towards the end, I'll add more error handling, but for now when we do `Map.get()` with a user that does not exist, it will return null. Also, for now, hopefully our system won't allow users that don't exist to access this method.